prover/src/shared_state.rs

```diff
@@ -189,21 +191,29 @@ macro_rules! gen_proof {
                 // aggregation_proof.proof = proof.into();
             } else {
                 let time_started = Instant::now();
-                let proof = gen_proof::<
-                    _,
-                    _,
-                    EvmTranscript<G1Affine, _, _, _>,
-                    EvmTranscript<G1Affine, _, _, _>,
-                    _,
-                >(
-                    &param,
-                    &pk,
-                    circuit,
-                    circuit_instance.clone(),
-                    fixed_rng(),
-                    task_options.mock_feedback,
-                    task_options.verify_proof,
-                );
+                let handle = tokio::task::spawn_blocking(move || {
+                    gen_proof::<
+                        _,
+                        _,
+                        EvmTranscript<G1Affine, _, _, _>,
+                        EvmTranscript<G1Affine, _, _, _>,
+                        _,
+                    >(
+                        &param,
+                        &pk,
+                        circuit,
+                        circuit_instance.clone(),
+                        fixed_rng(),
+                        task_options.mock_feedback,
+                        task_options.verify_proof,
+                    )
+                });
+
+                let proof = tokio::time::timeout(GEN_TIMEOUT, handle)
+                    .await
+                    .map_err(|e| e.to_string())?
```

Comment:
what does the error string look like? If david want to query in log.

Reply:
one of four:
1. "deadline has elapsed" when timeout
2. "task {id} was cancelled" when spawned gen task be cancelled
3. "task {id} panicked" when spawned gen task panic
4. other error from gen task